pkg/reconciler/serverlessservice/serverlessservice.go

```diff
@@ -209,48 +209,54 @@ func (r *reconciler) reconcilePublicEndpoints(ctx context.Context, sks *netv1alp
 	if err != nil {
 		return fmt.Errorf("failed to get private K8s Service endpoints: %w", err)
 	}
+
 	// We still might be "ready" even if in proxy mode,
 	// if proxy mode is by means of burst capacity handling.
 	pvtReady := presources.ReadyAddressCount(pvtEps)
-	if pvtReady > 0 {
-		foundServingEndpoints = true
+	sharedReady := presources.ReadyAddressCount(activatorEps)
+
+	logger.Infof("SKS is in %s mode; has %d endpoints in %s; %d activator endpoints",
+		sks.Spec.Mode, pvtReady, psn, sharedReady)
+
+	// Spew is expensive and there might be a lof of endpoints.
+	if dlogger.Core().Enabled(zap.DebugLevel) {
+		dlogger.Debug("Private endpoints: " + spew.Sprint(pvtEps.Subsets))
+		dlogger.Debug(fmt.Sprintf("Subset of activator endpoints (needed %d): %s",
+			sks.Spec.NumActivators, spew.Sprint(activatorEps)))
 	}
 
 	// The logic below is as follows:
+	// mode = sks.Spec.Mode
+	// if len(private_service_endpoints) == 0:
+	//    // Override the mode to Proxy if the service has no endpoints
+	//    mode = Proxy
+	// if len(activator_endpoints) < sks.Spec.NumActivators:
```

Comment:
Does this mean we go directly to the service even if TBC == -1?

Reply:
yes, but only if there are zero available activators

Reply:
That's not the logic, though as written. It is fewer than desired, which might be 4 < 5 :-)

Reply:
Yeah, I updated the comment.  This is what the logic was when I first added things, and reverted to just the `0` case because I didn't want to change the behavior in cases where the activator was simply underprovisioned because the HPA hadn't kicked in.